wagtail/blocks/base.py

```diff
@@ -94,6 +95,7 @@ def __init__(self, **kwargs):
         self.creation_counter = Block.creation_counter
         Block.creation_counter += 1
         self.definition_prefix = "blockdef-%d" % self.creation_counter
+        Block.definition_registry[self.definition_prefix] = self
```

Comment:
Not sure whether we'd like to use a different ID (or maybe even just use `creation_counter` as-is).

If we use `creation_counter` (as an integer), the view will need to safely convert `request.GET["id"]` back to an integer, which seems like an unnecessary hassle. Yet if we stringify the integer as the dict key, it doesn't seem right. So I opted to use `definition_prefix` for now.